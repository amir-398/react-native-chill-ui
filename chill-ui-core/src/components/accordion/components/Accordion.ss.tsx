import type { AccordionPropsSs } from '@types';

import { BoxSs } from '@components/box';
import { PropsWithChildren } from 'react';

import styles from '../styles/Accordion.ss.styles';
import { AccordionProvider } from './AccordionContext';
import { accordionDefaultProps } from '../utils/defaultProps';

/**
 * Accordion component provides a collapsible content area with support for single or multiple items.
 *
 * @example
 * ```tsx
 * <Accordion type="single" collapsible>
 *   <AccordionItem value="item-1">
 *     <AccordionTrigger>Is it accessible?</AccordionTrigger>
 *     <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
 *   </AccordionItem>
 * </Accordion>
 * ```
 *
 * @param children - Child AccordionItem components
 * @param type - Whether single or multiple items can be open at once ('single' | 'multiple')
 * @param collapseIcon - Icon to show when item is expanded (default: 'angle-down-solid')
 * @param collapsible - Whether items can be collapsed when type is 'single' (default: false)
 * @param defaultValue - Default open items (string for single, array for multiple)
 * @param disabled - Disable the entire accordion (default: false)
 * @param expandIcon - Icon to show when item is collapsed (default: 'angle-down-solid')
 * @param hasCollapseIcon - Whether to show collapse/expand icons (default: true)
 * @param iconPosition - Position of the expand/collapse icon (default: 'right')
 * @param onValueChange - Callback fired when accordion state changes
 * @param props - Additional ViewProps for the container
 *
 * @see {@link https://github.com/your-repo/chill-ui/tree/main/src/components/accordion/README.md Documentation}
 */
export default function Accordion(props: PropsWithChildren<AccordionPropsSs>) {
  const {
    children,
    collapseIcon = accordionDefaultProps.collapseIcon,
    collapsible = accordionDefaultProps.collapsible,
    defaultValue,
    disabled = accordionDefaultProps.disabled,
    expandIcon = accordionDefaultProps.expandIcon,
    hasCollapseIcon = accordionDefaultProps.hasCollapseIcon,
    iconPosition = accordionDefaultProps.iconPosition,
    onValueChange,
    style,
    type = accordionDefaultProps.type,
  } = props;

  return (
    <AccordionProvider
      type={type!}
      collapsible={collapsible}
      defaultValue={defaultValue}
      disabled={disabled}
      hasAnimation={false}
      animationDuration={0}
      hasCollapseIcon={hasCollapseIcon ?? true}
      iconPosition={iconPosition}
      expandIcon={expandIcon}
      collapseIcon={collapseIcon}
      onValueChange={onValueChange}
    >
      <BoxSs style={[styles.accordion, style]}>{children}</BoxSs>
    </AccordionProvider>
  );
}
