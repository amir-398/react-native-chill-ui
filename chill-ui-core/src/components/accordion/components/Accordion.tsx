import type { AccordionPropsTw } from '@types';

import { Box } from '@components/box';
import { PropsWithChildren } from 'react';
import { classNameHandler, cn, styleHandler } from '@utils';

import styles from '../styles/Accordion.styles';
import { AccordionProvider } from './AccordionContext';
import { accordionDefaultProps } from '../utils/defaultProps';
import { accordionClassName } from '../styles/Accordion.variants';
import { classNamePropsHandler } from '../../../utils/hybrid/classNamePropsHandler';

/**
 * Accordion component provides a collapsible content area with support for single or multiple items.
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * <Accordion type="single" collapsible>
 *   <AccordionItem value="item-1">
 *     <AccordionTrigger>Is it accessible?</AccordionTrigger>
 *     <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
 *   </AccordionItem>
 * </Accordion>
 * ```
 *
 * @param children - Child AccordionItem components
 * @param type - Whether single or multiple items can be open at once ('single' | 'multiple')
 * @param className - Custom CSS classes for styling
 * @param collapseIcon - Icon to show when item is expanded (default: 'angle-down-solid')
 * @param collapsible - Whether items can be collapsed when type is 'single' (default: false)
 * @param defaultValue - Default open items (string for single, array for multiple)
 * @param disabled - Disable the entire accordion (default: false)
 * @param expandIcon - Icon to show when item is collapsed (default: 'angle-down-solid')
 * @param hasCollapseIcon - Whether to show collapse/expand icons (default: true)
 * @param iconPosition - Position of the expand/collapse icon (default: 'right')
 * @param onValueChange - Callback fired when accordion state changes
 * @param props - Additional ViewProps for the container
 *
 * @see {@link https://github.com/your-repo/chill-ui/tree/main/src/components/accordion/README.md Documentation}
 */
export default function Accordion(props: PropsWithChildren<AccordionPropsTw>) {
  classNamePropsHandler(props, 'Accordion');
  const {
    children,
    className,
    collapseIcon = accordionDefaultProps.collapseIcon,
    collapsible = accordionDefaultProps.collapsible,
    defaultValue,
    disabled = accordionDefaultProps.disabled,
    expandIcon = accordionDefaultProps.expandIcon,
    hasCollapseIcon = accordionDefaultProps.hasCollapseIcon,
    iconPosition = accordionDefaultProps.iconPosition,
    onValueChange,
    style,
    type = accordionDefaultProps.type,
  } = props;

  return (
    <AccordionProvider
      type={type!}
      collapsible={collapsible}
      defaultValue={defaultValue}
      disabled={disabled}
      hasAnimation={false}
      animationDuration={0}
      hasCollapseIcon={hasCollapseIcon ?? true}
      iconPosition={iconPosition}
      expandIcon={expandIcon}
      collapseIcon={collapseIcon}
      onValueChange={onValueChange}
    >
      <Box
        {...classNameHandler(cn(accordionClassName, className))}
        {...styleHandler({ defaultStyle: styles.accordion, style })}
      >
        {children}
      </Box>
    </AccordionProvider>
  );
}
