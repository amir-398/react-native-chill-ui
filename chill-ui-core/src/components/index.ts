export * from './box';
export * from './toast';
export * from './loadingIndicatorsKit';
export * from './accordion';
export * from './dialog';
export * from './ripplePressable';
export * from './animatedBox';
export * from './icon';
export * from './avatar';
export * from './string';
export * from './inputDropdown';
export * from './AutocompleteDropdown';
export * from './highlightString';
export * from './buttonIcon';
export * from './checkbox';
export * from './chip';
export * from './maskedInput';
export * from './dropdownMenu';
export * from './inputSelectDropdown';
export * from './phoneNumberInput';
export * from './placesInput';
export * from './segmentedControl';
export * from './separator';
export * from './skeletons';
export * from './timePicker';
export * from './toggle';
export * from './wrapper';
