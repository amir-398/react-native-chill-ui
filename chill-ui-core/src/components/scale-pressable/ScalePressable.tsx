import { cn } from '@utils';
import { useRef, PropsWithChildren } from 'react';
import { Animated, Pressable, PressableProps } from 'react-native';

import { isNativeWindInstalled } from '@/utils';
import { classNamePropsHandler } from '@/utils/hybrid/classNamePropsHandler';

import { AnimatedBox } from '../animatedBox';

/**
 * Props for the ScalePressable component
 */
export interface ScalePressableProps extends Omit<PressableProps, 'style'> {
  /**
   * Custom CSS classes for styling the container (NativeWind)
   */
  className?: string;
  /**
   * Style object for the pressable container
   */
  style?: any;
  /**
   * Scale factor when pressed (default: 0.95)
   */
  scaleValue?: number;
  /**
   * Animation duration in milliseconds (default: 100)
   */
  duration?: number;
  /**
   * Callback function called when the component is pressed
   */
  onPress?: (event: any) => void;
}

/**
 * ScalePressable component that provides a scale effect on press.
 *
 * When pressed, the component scales down to give a tactile feedback effect,
 * making it feel like you're physically pressing a button.
 *
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * // Basic usage with scale effect
 * <ScalePressable onPress={() => console.log('Pressed!')}>
 *   <Box className="p-4 bg-blue-500 rounded-lg">
 *     <String color="white">Press me</String>
 *   </Box>
 * </ScalePressable>
 *
 * // With custom scale and duration
 * <ScalePressable
 *   scaleValue={0.9}
 *   duration={150}
 *   onPress={() => handleButtonPress()}
 * >
 *   <Box className="p-6 bg-green-500 rounded-xl">
 *     <String color="white" className="text-center font-bold">
 *       Custom Scale Button
 *     </String>
 *   </Box>
 * </ScalePressable>
 * ```
 *
 * @param children - Child components to render with scale effect (required)
 * @param className - Custom CSS classes for styling the container (NativeWind)
 * @param style - Style object for the pressable container
 * @param scaleValue - Scale factor when pressed (default: 0.95)
 * @param duration - Animation duration in milliseconds (default: 100)
 * @param onPress - Callback function called when the component is pressed
 * @returns ScalePressable component with scale animation
 * @throws Error if no children are provided
 */
export default function ScalePressable(props: PropsWithChildren<ScalePressableProps>) {
  classNamePropsHandler(props, 'ScalePressable');
  const { children, className, duration = 100, onPress, scaleValue = 0.95, style, ...rest } = props;

  if (!children) {
    throw new Error('ScalePressable must have children');
  }

  const scaleAnim = useRef(new Animated.Value(1)).current;

  const handlePressIn = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: scaleValue,
      useNativeDriver: true,
    }).start();
  };

  const handlePressOut = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: 1,
      useNativeDriver: true,
    }).start();
  };

  const handlePress = (event: any) => {
    if (onPress) {
      onPress(event);
    }
  };

  const animatedStyle = {
    transform: [{ scale: scaleAnim }],
  };

  return isNativeWindInstalled() ? (
    <AnimatedBox style={animatedStyle}>
      <Pressable
        onPress={handlePress}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
        // @ts-ignore
        className={cn('self-start', className)}
        style={style}
        {...rest}
      >
        {children}
      </Pressable>
    </AnimatedBox>
  ) : (
    <AnimatedBox style={animatedStyle}>
      <Pressable
        onPress={handlePress}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
        style={[{ alignSelf: 'flex-start' }, style]}
        {...rest}
      >
        {children}
      </Pressable>
    </AnimatedBox>
  );
}
