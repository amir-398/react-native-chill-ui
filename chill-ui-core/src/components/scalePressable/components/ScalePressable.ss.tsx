import { ScalePressablePropsSs } from '@types';
import { Animated, Pressable } from 'react-native';
import { AnimatedBoxSs } from '@components/animatedBox';
import { useRef, PropsWithChildren, forwardRef } from 'react';

import { styles } from '../styles/ScalePressable.ss.styles';
import { scalePressableDefaultProps } from '../utils/defaultProps';

/**
 * ScalePressable component that provides a scale effect on press.
 *
 * When pressed, the component scales down to give a tactile feedback effect,
 * making it feel like you're physically pressing a button.
 *
 * @example
 * ```tsx
 * // Basic usage with scale effect
 * <ScalePressable onPress={() => console.log('Pressed!')}>
 *   <Box style={{ padding: 16, backgroundColor: '#3B82F6', borderRadius: 8 }}>
 *     <String color="white">Press me</String>
 *   </Box>
 * </ScalePressable>
 * ```
 *
 * @param children - Child components to render with scale effect (required)
 * @param style - Style object for the pressable container
 * @param scaleValue - Scale factor when pressed (default: 0.95)
 * @param duration - Animation duration in milliseconds (default: 100)
 * @param onPress - Callback function called when the component is pressed
 * @returns ScalePressable component with scale animation
 * @throws Error if no children are provided
 */
export const ScalePressable = forwardRef<any, PropsWithChildren<ScalePressablePropsSs>>((props, ref) => {
  const {
    children,
    duration = scalePressableDefaultProps.duration,
    onPress,
    scaleValue = scalePressableDefaultProps.scaleValue,
    style,
    ...rest
  } = props;

  const scaleAnim = useRef(new Animated.Value(1)).current;

  const handlePressIn = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: scaleValue,
      useNativeDriver: true,
    }).start();
  };

  const handlePressOut = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: 1,
      useNativeDriver: true,
    }).start();
  };

  const handlePress = (event: any) => {
    if (onPress) {
      onPress(event);
    }
  };

  const animatedStyle = {
    transform: [{ scale: scaleAnim }],
  };

  return (
    <AnimatedBoxSs style={animatedStyle}>
      <Pressable
        ref={ref}
        onPress={handlePress}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
        style={[styles.container, style]}
        {...rest}
      >
        {children}
      </Pressable>
    </AnimatedBoxSs>
  );
});

ScalePressable.displayName = 'ScalePressable';
