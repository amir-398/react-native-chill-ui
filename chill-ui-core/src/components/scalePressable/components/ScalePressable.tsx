import { ScalePressablePropsTw } from '@types';
import { Animated, Pressable } from 'react-native';
import { AnimatedBox } from '@components/animatedBox';
import { useRef, PropsWithChildren, forwardRef } from 'react';
import { classNameHandler, cn, classNamePropsHandler, styleHandler } from '@utils';

import { styles } from '../styles/ScalePressable.ss.styles';
import { twStyles } from '../styles/ScalePressable.tw.styles';
import { scalePressableDefaultProps } from '../utils/defaultProps';

/**
 * ScalePressable component that provides a scale effect on press.
 *
 * When pressed, the component scales down to give a tactile feedback effect,
 * making it feel like you're physically pressing a button.

 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * // Basic usage with scale effect
 * <ScalePressable onPress={() => console.log('Pressed!')}>
 *   <Box className="p-4 bg-blue-500 rounded-lg">
 *     <String color="white">Press me</String>
 *   </Box>
 * </ScalePressable>
 * ```
 *
 * @param children - Child components to render with scale effect (required)
 * @param className - Custom CSS classes for styling the container (NativeWind)
 * @param style - Style object for the pressable container
 * @param scaleValue - Scale factor when pressed (default: 0.95)
 * @param duration - Animation duration in milliseconds (default: 100)
 * @param onPress - Callback function called when the component is pressed
 * @returns ScalePressable component with scale animation
 * @throws Error if no children are provided
 */
export const ScalePressable = forwardRef<any, PropsWithChildren<ScalePressablePropsTw>>((props, ref) => {
  classNamePropsHandler(props, 'ScalePressable');
  const {
    children,
    className,
    duration = scalePressableDefaultProps.duration,
    onPress,
    scaleValue = scalePressableDefaultProps.scaleValue,
    style,
    ...rest
  } = props;

  const scaleAnim = useRef(new Animated.Value(1)).current;

  const handlePressIn = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: scaleValue,
      useNativeDriver: true,
    }).start();
  };

  const handlePressOut = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: 1,
      useNativeDriver: true,
    }).start();
  };

  const handlePress = (event: any) => {
    if (onPress) {
      onPress(event);
    }
  };

  const animatedStyle = {
    transform: [{ scale: scaleAnim }],
  };

  return (
    <AnimatedBox style={animatedStyle}>
      <Pressable
        ref={ref}
        onPress={handlePress}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
        {...classNameHandler(cn(twStyles.container, className))}
        {...styleHandler({ defaultStyle: styles.container, style })}
        {...rest}
      >
        {children}
      </Pressable>
    </AnimatedBox>
  );
});

ScalePressable.displayName = 'ScalePressable';
