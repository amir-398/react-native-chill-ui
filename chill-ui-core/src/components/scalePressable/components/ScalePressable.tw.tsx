import { cn } from '@utils';
import { ScalePressablePropsTw } from '@types';
import { Animated, Pressable } from 'react-native';
import { AnimatedBoxTw } from '@components/animatedBox';
import { useRef, PropsWithChildren, forwardRef } from 'react';

import { twStyles } from '../styles/ScalePressable.tw.styles';
import { scalePressableDefaultProps } from '../utils/defaultProps';

/**
 * ScalePressable component that provides a scale effect on press.
 *
 * When pressed, the component scales down to give a tactile feedback effect,
 * making it feel like you're physically pressing a button.
 *
 * @example
 * ```tsx
 * // Basic usage with scale effect
 * <ScalePressable onPress={() => console.log('Pressed!')}>
 *   <Box className="p-4 bg-blue-500 rounded-lg">
 *     <String color="white">Press me</String>
 *   </Box>
 * </ScalePressable>
 * ```
 *
 * @param children - Child components to render with scale effect (required)
 * @param className - Custom CSS classes for styling the container (NativeWind)
 * @param style - Style object for the pressable container
 * @param scaleValue - Scale factor when pressed (default: 0.95)
 * @param duration - Animation duration in milliseconds (default: 100)
 * @param onPress - Callback function called when the component is pressed
 * @returns ScalePressable component with scale animation
 * @throws Error if no children are provided
 */
export const ScalePressable = forwardRef<any, PropsWithChildren<ScalePressablePropsTw>>((props, ref) => {
  const {
    children,
    className,
    duration = scalePressableDefaultProps.duration,
    onPress,
    scaleValue = scalePressableDefaultProps.scaleValue,
    style,
    ...rest
  } = props;

  const scaleAnim = useRef(new Animated.Value(1)).current;

  const handlePressIn = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: scaleValue,
      useNativeDriver: true,
    }).start();
  };

  const handlePressOut = () => {
    Animated.timing(scaleAnim, {
      duration,
      toValue: 1,
      useNativeDriver: true,
    }).start();
  };

  const handlePress = (event: any) => {
    if (onPress) {
      onPress(event);
    }
  };

  const animatedStyle = {
    transform: [{ scale: scaleAnim }],
  };

  return (
    <AnimatedBoxTw style={animatedStyle}>
      <Pressable
        ref={ref}
        onPress={handlePress}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
        className={cn(twStyles.container, className)}
        style={style}
        {...rest}
      >
        {children}
      </Pressable>
    </AnimatedBoxTw>
  );
});

ScalePressable.displayName = 'ScalePressable';
