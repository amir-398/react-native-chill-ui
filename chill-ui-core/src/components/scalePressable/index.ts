export { ScalePressable } from './components/ScalePressable';
export { ScalePressable as ScalePressableTw } from './components/ScalePressable.tw';
export { ScalePressable as ScalePressableSs } from './components/ScalePressable.ss';
