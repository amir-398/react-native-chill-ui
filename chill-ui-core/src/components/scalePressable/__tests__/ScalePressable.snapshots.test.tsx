import { Text, View } from 'react-native';
import { render } from '@testing-library/react-native';

import { ScalePressable } from '../components/ScalePressable';

// Mock utils
jest.mock('../../../utils', () => ({
  classNameHandler: jest.fn(() => ({})),
  classNamePropsHandler: jest.fn(),
  cn: jest.fn((...args) => args.filter(Boolean).join(' ')),
  colorVariantPropsHandler: jest.fn(),
  isString: jest.fn(value => typeof value === 'string'),
  styleHandler: jest.fn(() => ({})),
}));

// Mock AnimatedBox
jest.mock('../../../components/animatedBox', () => ({
  AnimatedBox: ({ children, ...props }: any) => {
    const { View: ReactNativeView } = require('react-native');
    return (
      <ReactNativeView testID="animated-box" {...props}>
        {children}
      </ReactNativeView>
    );
  },
}));

describe('ScalePressable Snapshots', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot with basic props', () => {
    const tree = render(
      <ScalePressable>
        <Text>Basic Content</Text>
      </ScalePressable>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom scaleValue', () => {
    const tree = render(
      <ScalePressable scaleValue={0.85}>
        <Text>Small Scale</Text>
      </ScalePressable>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom duration', () => {
    const tree = render(
      <ScalePressable duration={200}>
        <Text>Fast Animation</Text>
      </ScalePressable>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with disabled state', () => {
    const tree = render(
      <ScalePressable disabled>
        <Text>Disabled</Text>
      </ScalePressable>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom styling', () => {
    const tree = render(
      <ScalePressable className="custom-class" style={{ margin: 10 }}>
        <View style={{ backgroundColor: '#10B981', borderRadius: 8, padding: 16 }}>
          <Text style={{ color: 'white' }}>Styled Button</Text>
        </View>
      </ScalePressable>,
    );
    expect(tree).toMatchSnapshot();
  });
});
