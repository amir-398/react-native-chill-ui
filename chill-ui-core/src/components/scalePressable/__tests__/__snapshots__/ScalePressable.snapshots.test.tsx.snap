// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScalePressable Snapshots should match snapshot with basic props 1`] = `
<View
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-box"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Basic Content
    </Text>
  </View>
</View>
`;

exports[`ScalePressable Snapshots should match snapshot with custom duration 1`] = `
<View
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-box"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Fast Animation
    </Text>
  </View>
</View>
`;

exports[`ScalePressable Snapshots should match snapshot with custom scaleValue 1`] = `
<View
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-box"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Small Scale
    </Text>
  </View>
</View>
`;

exports[`ScalePressable Snapshots should match snapshot with custom styling 1`] = `
<View
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-box"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "backgroundColor": "#10B981",
          "borderRadius": 8,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        Styled Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`ScalePressable Snapshots should match snapshot with disabled state 1`] = `
<View
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-box"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Disabled
    </Text>
  </View>
</View>
`;
